Automattic/newspack-ads
# fix(gam): validate api session

This PR ensures that the API session is valid before attempting to fetch objects.

### How to test

1. Make sure you are on the **master** branch
2. Connect through Service Account or OAuth with an account that doesn't have access to any GAM account
3. Visit the Advertising wizard and confirm the fatal error
4. Visit the WP's Customizer and also confirm the fatal error
5. Check out this branch, refresh the wizard and the Customizer
6. Confirm there are no fatal errors and GAM is being treated as disconnected/unavailable

In the future, we can handle messaging to notify that the session doesn't connect to any GAM account. For now, let's just avoid the fatal errors.
